compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Error.hs

```diff
@@ -385,9 +387,37 @@ instance Pretty Error where
       "tried to perform key lookup or fetch on template " <> pretty tpl
     EExpectedOptionalType typ -> do
       "expected list type, but found: " <> pretty typ
-    EExpectedViewType actuallyFound typ -> do
+    EViewTypeHeadNotCon badHead typ ->
```

Comment:
I like that all the stringy logic lives here now, but I don't like the duplication (all three error messages start with `"expected monomorphic record type in view type, but found "` and include the same explanation of record types). That's not a blocker, though, I think we could fix that as part of https://github.com/digital-asset/daml/issues/14690